cpp/src/hash/hashing.cu

```diff
@@ -38,6 +38,22 @@ bool md5_type_check(data_type dt)
   return !is_chrono(dt) && (is_fixed_width(dt) || (dt.id() == type_id::STRING));
 }
 
+template <typename IterType>
+std::vector<column_view> to_leaf_columns(IterType iter_begin, IterType iter_end)
+{
+  std::vector<column_view> leaf_columns;
+  for_each(iter_begin, iter_end, [&](column_view const& col) {
+    if (is_nested(col.type())) {
+      CUDF_EXPECTS(col.type().id() == type_id::STRUCT, "unsupported nested type");
+      auto child_columns = to_leaf_columns(col.child_begin(), col.child_end());
+      leaf_columns.insert(leaf_columns.end(), child_columns.begin(), child_columns.end());
+    } else {
+      leaf_columns.emplace_back(col);
+    }
+  });
```

Comment:
This looks like a `transform`

Reply:
I don't think `transform` applies here because that takes a unary operation and performs a one-to-one mapping from the input iterator to the output iterator.  This isn't a one-to-one mapping of inputs to outputs.  A single input can map to multiple outputs when a struct is encountered (implemented via the recursion).

Reply:
Ah, you are definitely correct! Sorry, I missed that `insert` was doing multiple insertions.